nis/src/test/java/org/nem/nis/ForkConfigurationTest.java

```diff
@@ -92,26 +129,40 @@ public void canCreateCustomConfiguration() {
 		final List<Hash> fallbackHashes = Arrays.stream(fallbackHashStrings).map(Hash::fromHexString).collect(Collectors.toList());
 
 		// Act:
-		final ForkConfiguration config = new ForkConfiguration(new BlockHeight(1234), hashes, fallbackHashes);
+		final ForkConfiguration config = new ForkConfiguration.Builder().treasuryReissuanceForkHeight(new BlockHeight(1234))
+				.treasuryReissuanceForkTransactionHashes(hashes).treasuryReissuanceForkFallbackTransactionHashes(fallbackHashes)
+				.feeForkHeight(new BlockHeight(1)).mosaicsForkHeight(new BlockHeight(2)).multisigMOfNForkHeight(new BlockHeight(3))
+				.remoteAccountForkHeight(new BlockHeight(4)).mosaicRedefinitionForkHeight(new BlockHeight(5))
+				.secondFeeForkHeight(new BlockHeight(6)).build();
 
 		// Assert:
 		MatcherAssert.assertThat(config.getTreasuryReissuanceForkHeight(), IsEqual.equalTo(new BlockHeight(1234)));
 		MatcherAssert.assertThat(config.getTreasuryReissuanceForkTransactionHashes(), IsEqual.equalTo(hashes));
 		MatcherAssert.assertThat(config.getTreasuryReissuanceForkFallbackTransactionHashes(), IsEqual.equalTo(fallbackHashes));
+
+		MatcherAssert.assertThat(config.getFeeForkHeight(), IsEqual.equalTo(new BlockHeight(1)));
+		MatcherAssert.assertThat(config.getMosaicsForkHeight(), IsEqual.equalTo(new BlockHeight(2)));
+		MatcherAssert.assertThat(config.getmultisigMOfNForkHeight(), IsEqual.equalTo(new BlockHeight(3)));
+		MatcherAssert.assertThat(config.getRemoteAccountForkHeight(), IsEqual.equalTo(new BlockHeight(4)));
+		MatcherAssert.assertThat(config.getMosaicRedefinitionForkHeight(), IsEqual.equalTo(new BlockHeight(5)));
+		MatcherAssert.assertThat(config.getSecondFeeForkHeight(), IsEqual.equalTo(new BlockHeight(6)));
 	}
 
 	@Test
 	public void canReadDefaultConfiguration() {
 		// Arrange:
-		final Properties properties = new Properties();
+		final int version = NetworkInfos.getDefault().getVersion() << 24;
 
-		// Act:
-		final ForkConfiguration config = new ForkConfiguration(new NemProperties(properties));
+		canReadConfiguration(version, () -> new ForkConfiguration.Builder(new NemProperties(new Properties())).build());;
+	}
 
-		// Assert:
-		MatcherAssert.assertThat(config.getTreasuryReissuanceForkHeight(), IsEqual.equalTo(new BlockHeight(1)));
-		MatcherAssert.assertThat(config.getTreasuryReissuanceForkTransactionHashes(), IsEqual.equalTo(new ArrayList<Hash>()));
-		MatcherAssert.assertThat(config.getTreasuryReissuanceForkFallbackTransactionHashes(), IsEqual.equalTo(new ArrayList<Hash>()));
+	@Test
+	public void canReadMainnetConfiguration() {
+		// Arrange:
+		final NetworkInfo networkInfo = NetworkInfos.getMainNetworkInfo();
+		final int version = networkInfo.getVersion() << 24;
+
+		canReadConfiguration(version, () -> new ForkConfiguration.Builder(new NemProperties(new Properties()), networkInfo).build());
```

Comment:
should there be separate ctor that accepts only networkInfo ?

when would well-defined network properties need to be overridden?

Reply:
currently the network is only overridden from the nis configuration.  So the Nem properties are also available.

Yea, can create a ctor that only accepts networkinfo